app/src/lib/promise.ts

```diff
@@ -7,35 +7,13 @@
  *
```

Comment:
It's not related to this PR, but the comment in L6 seems to be incomplete, maybe we can fix it while we're modifying this file.